src/player_hardcoded_effects.cpp

```diff
@@ -1190,29 +1188,6 @@ static void eff_fun_sleep( Character &u, effect &it )
                 }
             }
         }
-        if( u.has_trait( trait_SCHIZOPHRENIC ) && !u.has_effect( effect_took_thorazine ) &&
-            one_in( 43200 ) && u.is_avatar() ) {
-            if( one_in( 2 ) ) {
-                u.sound_hallu();
-            } else {
-                int max_count = rng( 1, 3 );
-                int count = 0;
-                for( const tripoint &mp : here.points_in_radius( u.pos(), 1 ) ) {
-                    if( mp == u.pos() ) {
-                        continue;
-                    }
-                    if( here.has_flag( ter_furn_flag::TFLAG_FLAT, mp ) &&
-                        here.pl_sees( mp, 2 ) ) {
-                        g->spawn_hallucination( mp );
-                        if( ++count > max_count ) {
-                            break;
-                        }
-                    }
-                }
-            }
-            it.set_duration( 0_turns );
-            woke_up = true;
-        }
```

Comment:
was this part ported to json, because i can't find anything that can spawn hallucinations in the `hallucination_eocs.json`

Reply:
I didn't port this to JSON, I just used the hallucinating effect to do that.

Reply:
isn't hallucination effect is what was jsonified here?
i just tested, applied hallu, and got no hallucinations (aka hallucination of monsters that tries to attack you) at all, and i assume this piece of code respond for it